Specialist finder for MOD (WHIT-1981)

ChrisBAshton opened the pull request:
## What

Create a finder for MOD armed forces covenant pledges, originally here https://www.gov.uk/government/publications/armed-forces-corporate-covenant-signed-pledges.

Data will be imported via a rake task - see [PR here](https://github.com/alphagov/specialist-publisher/pull/3330).

JIRA: https://gov-uk.atlassian.net/browse/WHIT-1981

## Testing

Running the rake task took around an hour locally, all rows were imported successfully.

<img width="765" height="489" alt="Screenshot 2025-08-13 at 14 16 43" src="https://github.com/user-attachments/assets/c389ab68-6519-4d6d-bd12-bcc9d23d2176" />
<img width="642" height="953" alt="Screenshot 2025-08-13 at 14 16 07" src="https://github.com/user-attachments/assets/6966bd4f-eef4-4a5b-809d-b21b01d1d905" />
<img width="667" height="933" alt="Screenshot 2025-08-13 at 14 16 31" src="https://github.com/user-attachments/assets/7306f1ce-4af2-4690-af85-8eefd26cd53a" />
<img width="1026" height="782" alt="Screenshot 2025-08-13 at 14 15 36" src="https://github.com/user-attachments/assets/5d50267d-bf87-4b2d-a500-cb279e0d84cb" />
<img width="1158" height="841" alt="Screenshot 2025-08-12 at 20 43 19" src="https://github.com/user-attachments/assets/421bcd0e-6b35-45c4-a45c-5f4b7cd516ef" />
<img width="503" height="629" alt="Screenshot 2025-08-13 at 18 31 56" src="https://github.com/user-attachments/assets/6d974f18-2c31-462a-9783-120541785353" />
<img width="503" height="517" alt="Screenshot 2025-08-13 at 18 31 15" src="https://github.com/user-attachments/assets/e53de0de-c2f9-4d94-93b3-66e6ee0f8b94" />

ChrisBAshton committed:
Make long fields (such as 'Body template') wrap on schema screen

Otherwise long templates require lots of horizontal scrolling.

ChrisBAshton committed:
Remove line about 'corresponding view'

No new finder should have its own bespoke view.

ChrisBAshton committed:
Add 'Businesses who have signed the Armed Forces Covenant' finder

JIRA: https://gov-uk.atlassian.net/browse/WHIT-1981

ChrisBAshton committed:
Fix key generation

The output of `Date signed` as a field name, was `datesigned` as
a key. The `.underscore` wasn't doing what we might hope!

The reason this worked in the tests is that each word was
capitalised. The `underscore` method simply changes strings from
CamelCase to snake_case.

Whilst we're here, we have a number of values that use ampersands
and it would be nice to have these represented in the key if possible,
so now convert `&` to `and`.

And we also had a bug whereby any name field that had a comma in
it would have the comma preserved in the generated key. So we're
now stripping all invalid characters and leading/trailing underscores.

ChrisBAshton committed:
Refactor label/value/key derivation

The wrong abstractions were being used here, making the code
rather difficult to follow. Splitting out the 'extract value' and
'extract label' methods made it much easier to reason about, and
I've consolidated the behaviour that converts an arbitrary string
to a key-like string (with explicit 'separator' parameter of either
"-" or "_" accordingly).

lauraghiorghisor-tw commented on app/models/facet.rb:
```diff
@@ -53,7 +53,11 @@ def from_finder_admin_form_params(params)
 
     def facet_key(key, name)
-      key.presence || name&.gsub(" ", "")&.underscore
+      key.presence || name&.gsub(" ", "_")
```
Ah, nice, this has been bugging me for a while ❤️

lauraghiorghisor-tw committed:
Fix render syntax

Triggered this render during development, and got "You invoked render but did not give any of :body, :file, :html, :inline, :partial, :plain, :renderable, or :template option." - fixed by using render with "partial" instead.

lauraghiorghisor-tw committed:
Update documentation

- Testing:
Remove not very useful commands on `grep`-ing tests. This only filters the output, test suite still runs all tests. I think we've fixed the format dependency - bespoke tests are only for custom views and validations.

- Correct instructions on "format":
The line previously suggested that you should add the "format" to publishing api (which could mean the `filter.format`), but it's actually the document type that we use. Linked to the naming conventions readme. There was an attempt to decouple these notions, or make the usage more expressive, but it was closed - here https://github.com/alphagov/specialist-publisher/pull/3166

- Add link to instructions on running finder frontend

TonyGDS reviewed (approved):
LGTM
